source/modules/runners/run.py

```diff
@@ -131,45 +131,47 @@ def run_playitem(self, config: dict, options: dict, inventory: str, lockpath: st
             for playitem in playitems:
                 command, command_env = self.construct_command(playitem, inventory, config, options)
                 self.logger.debug("Running '%s'.", command)
-                try:
-                    with subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
-                                          env=command_env) as proc:
-
-                        for msg in proc.stdout:
-                            dec_msg = msg.split(b"\n")[0].decode("utf-8")
-                            returned.append(dec_msg)
-                            if options["raw_output"]:
-                                print(dec_msg)
-
-                        proc.communicate()
-                        format_obj = Formatters(self.logger)
-                        parsed_std = format_obj.format_ansible_output(returned)
-                        host_list = db_writer.parse_yaml_output_for_hosts(parsed_std["complete"],
-                                                                self.sequence_id)
-                        sequence_records = db_writer.start_sequence_dict(host_list,
-                                                                         self.setup_hooks, options,
-                                                                         self.start_ts_raw,
+                with open(f"{self.log_path}.raw", "a", encoding="utf-8") as raw_log:
```

Comment:
I would prefer to have a single point for logging and message redistribution. Is it really required at this moment? I mean, we can merge it but I will change this once I (finally? hopefully next week?) get back to ansible-deployer development.

Reply:
for me it was very helpful last days, the whole part of code is probably something we have to fully restructure to run as ansible callback plugin, so it will be a bigger change